### PR TITLE
OHSH-520 add initial private ca documentation

### PR DESCRIPTION
**Pull request by avanti-joshi**
# [Document how to use the Private Certificate Authority in AWS](https://ocio-jira.acf.hhs.gov/browse/OHSH-520)

Changes proposed in this pull request:

- Adding documentation around how to use Private CA to create certificates

### CODE REVIEW

File: docs/how-we-work/private-ca.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Terraform supports importing a aws_acmpca_certificate_authority resource as a datatype by it's arn. An example of this is below:
+
+```
+data "aws_acmpca_certificate_authority" "example" {
+  arn = "arn:aws:acm-pca:us-east-1:123456789012:certificate-authority12345678-012"
+}
+```
+
+More information about importing a private CA via terraform can be found [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/acmpca_certificate_authority)
```
Comment (rpdelaney) [inline docs/how-we-work/private-ca.md:29]:
```suggestion
More information about importing a private CA via terraform can be found [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/acmpca_certificate_authority).
```

File: docs/how-we-work/private-ca.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Without any modifications to your local machine "trust" settings, navigating to a website that uses a certificate signed by the OHS Private CA will result in certificate errors. To resolve these, you must download and save a copy of the root CA cert on your local machine, and "trust" it. The root CA certificate can be found on AWS in the ACM console.
+
+For Mac OS run the following:
+
+```
+sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ~/new-root-certificate.crt
+```
+
+For Windows:
```
Comment (rpdelaney) [inline docs/how-we-work/private-ca.md:49]: Do we have a use case for doing this on Windows? If not, omitting the `certutil` stuff may make the document a faster read.

File: docs/how-we-work/private-ca.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+## Private Certificate Authority (CA) Background and Creation
+
+The OHS private CA was created using the [acmpca certificate authority](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_certificate_authority) terraform module. Note that as stated in the terraform module, a the manual step of installing the root CA certificate from the console is required to change the status of the CA from pending to active. This is a one time action that only has to be done when creating a new Private CA.
```
Comment (rpdelaney) [inline docs/how-we-work/private-ca.md:3]:
```suggestion
The OHS private CA was created using the [ACM Private Certificate Authority](https://docs.aws.amazon.com/acm-pca/latest/userguide/PcaWelcome.html) Terraform [module](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_certificate_authority). Note that as stated in the Terraform module, a the manual step of installing the root CA certificate from the console is required to change the status of the CA from pending to active. This is a one time action that only has to be done when creating a new Private CA.
```

**Review by rpdelaney — APPROVED** [review]
Approved with comments - please make changes and merge.

This is a great start. Good job!

File: docs/how-we-work/private-ca.md (hunk 2 above)
Comment (avanti-joshi) [reply docs/how-we-work/private-ca.md:49]: @rpdelaney the reason I included it was incase we needed it when sending instructions to the app teams / jenkins users. Wasn't sure if all of OHS has Macs

Comment (avanti-joshi) [reply docs/how-we-work/private-ca.md:49]: if we have confirmation on that though I definitely agree it doesn't need to be included

Comment (rpdelaney) [reply docs/how-we-work/private-ca.md:49]: Ahh yeah good thinking. Works for me.

File: docs/how-we-work/private-ca.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+## Private Certificate Authority (CA) Background and Creation
+
+The OHS private CA was created using the [ACM Private Certificate Authority](https://docs.aws.amazon.com/acm-pca/latest/userguide/PcaWelcome.html) Terraform [module](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_certificate_authority). Note that as stated in the Terraform module, a the manual step of installing the root CA certificate from the console is required to change the status of the CA from pending to active. This is a one time action that only has to be done when creating a new Private CA.
```
Comment (kilbergr) [inline docs/how-we-work/private-ca.md:3]:
```suggestion
The OHS private CA was created using the [ACM Private Certificate Authority](https://docs.aws.amazon.com/acm-pca/latest/userguide/PcaWelcome.html) Terraform [module](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_certificate_authority). Note that as stated in the Terraform module, the manual step of installing the root CA certificate from the console is required to change the status of the CA from pending to active. This is a one time action that only has to be done when creating a new Private CA.
```

File: docs/how-we-work/private-ca.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  certificate_authority_arn = aws_acmpca_certificate_authority.private_ca.arn
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+```
+
+## Importing Private CA For Use in Terraform
+
+Terraform supports importing a aws_acmpca_certificate_authority resource as a datatype by it's arn. An example of this is below:
```
Comment (kilbergr) [inline docs/how-we-work/private-ca.md:21]:
```suggestion
Terraform supports importing a aws_acmpca_certificate_authority resource as a datatype by its arn. An example of this is below:
```

Comment (kilbergr) [reply docs/how-we-work/private-ca.md:21]: Do we want to be importing them as a data type? I would imagine if we're pulling in to manage with tf, we'd want to import as the resource. Am I misunderstanding the use case?

File: docs/how-we-work/private-ca.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+A certificate can be referenced / used with it's arn:
+
+```
+certificate_arn   = aws_acm_certificate.cert_name.arn
+```
+
+## Resolving Cert Errors
+
+Without any modifications to your local machine "trust" settings, navigating to a website that uses a certificate signed by the OHS Private CA will result in certificate errors. To resolve these, you must download and save a copy of the root CA cert on your local machine, and "trust" it. The root CA certificate can be found on AWS in the ACM console.
```
Comment (kilbergr) [inline docs/how-we-work/private-ca.md:41]: Can you recommend a location to save the cert?

**Review by kilbergr — CHANGES_REQUESTED** [review]
Looking very good but I do have a few minor questions.

**Review by kilbergr — APPROVED** [review]
TY for changes!

**Pull request merged**